### ISSUE TITLE
linter: bun oxlint still loads oxlint.config.ts through Node and fails with ERR_UNKNOWN_FILE_EXTENSION in CI

### ISSUE THREAD

**Issue by osnoser1** [oxc-project/oxc#20303]
### What version of Oxlint are you using?

1.53.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```typescript
import { defineConfig } from 'oxlint';
import baseConfig from './oxlint.base.ts';

export default defineConfig({
  extends: [baseConfig],
  options: {
    typeAware: true,
    typeCheck: true,
  },
  // additional rules omitted for brevity
});
```

### What happened?

```
Failed to parse oxlint configuration file.

× Failed to load config: /home/runner/work/.../oxlint.config.ts
│
│ TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for /home/runner/work/.../oxlint.config.ts
│     at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:189:9)
│     at defaultGetFormat (node:internal/modules/esm/get_format:232:36)
│     at defaultLoad (node:internal/modules/esm/load:145:22)
│     at async ModuleLoader.loadAndTranslate (node:internal/modules/esm/loader:543:45)
│     at async ModuleJob._link (node:internal/modules/esm/module_job:148:19)

```

Hi, I'm not completely sure if this is an issue of `oxlint` or `bun`, but I expected `oxlint` to work with `bun` and with `oxlint.config.ts`, or at least not route config loading through a Node path that cannot load the .ts config.
What I suppose is that because `node_modules/.bin/oxlint` starts with `#!/usr/bin/env node` the CI uses the default node.js version is installed in the Ubuntu 24.04 ARM64 that doesn't have support to native ts

The workaround I did is just run the file directly with bun using the `package.json`: `bun ./node_modules/oxlint/bin/oxlint`

Either way, if any other person has this issue at least will have a way to apply a workaround

This issue happened in the CI

**Environment**
CI:
- GitHub Actions
- Ubuntu 24.04 ARM64
- Bun 1.3.9

Btw, thanks for this awesome project ❤️

**Comment by camc314** [comment]
> TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" 

this means you're using a node version that doesn't support loading typescript natively.

To fix this, move to `>=22.18.0` or `>=24.3.0` or `25.2.0`

If you wan to forcable use bun with oxlint, run `--bun` https://bun.com/docs/pm/bunx#param-bun. (bun invokes oxlint which has a hashbang at the top of the file that specifies to us node - supplying `--bun` overrides this behaviour and uses bun)

**Comment by osnoser1** [comment]
Yes, that is the idea, use bun instead of node
Thank you @camc314
